internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -574,10 +573,23 @@ func (c *container) mountImage(mnt *mount.Point) error {
 	}
 
 	path := fmt.Sprintf("/dev/loop%d", number)
-	sylog.Debugf("Mounting loop device %s to %s\n", path, mnt.Destination)
-	_, err = c.rpcOps.Mount(path, mnt.Destination, mnt.Type, flags, optsString)
-	if err != nil {
-		return fmt.Errorf("failed to mount %s filesystem: %s", mnt.Type, err)
+
+	sylog.Debugf("Mounting loop device %s to %s of type %s\n", path, mnt.Destination, mnt.Type)
+
+	if mnt.Type == "encryptfs" {
+		sp := strings.Split(path, "/")
+		loopdev := sp[len(sp)-1]
+		_, err = c.rpcOps.Decrypt(offset, loopdev)
+
+		_, err = c.rpcOps.Mount("/dev/mapper/sycrypt", mnt.Destination, "ext3", flags, optsString)
```

Comment:
Right now the device is hard-coded to "sycrypt" but I want to change it to dynamically generate a name.